dali/pipeline/util/bounding_box_utils.h

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.c++ copyswa
+
+#ifndef DALI_PIPELINE_UTIL_BOUNDING_BOX_UTILS_H_
+#define DALI_PIPELINE_UTIL_BOUNDING_BOX_UTILS_H_
+
+#include <limits>
+#include "dali/core/geom/box.h"
+#include "dali/core/tensor_layout.h"
+#include "dali/core/math_util.h"
+
+namespace dali {
+
+template <int ndim>
+void AxisFlip(Box<ndim, float>& box, int axis) {
```

Comment:
Are there unit-tests for the functions in this file? I know, some functions are trivial, but seems to me, that for e.g. `PermuteCoords` would be nice to have them

Reply:
to be considered in follow up PR (see my explanation below)